src/guides/resolve-zil-domain.md

```diff
@@ -0,0 +1,418 @@
+* **Record is not found** -&gt; This can be useful if you are querying domain for specific records and it is null
+
+Although any string can be stored as a key under the domain, Unstoppable domains [standardized some of the keys](https://docs.unstoppabledomains.com/domain-registry-essentials/records-reference) across many applications.   
+
+For the record is not found error message we are going to check if the domain has a BTC address and if not we will show the error without cleaning the entire DOM  
+
+```typescript
+type Resolution = {
+  resolverAddress: string,
+  ownerAddress: string,
+  records: {
+    [key in string]: string}
+  };
+
+function displayResolution(resolution: Resolution) {
+  const {ownerAddress, resolverAddress, records} = resolution;
+  const mainContainer = document.getElementById('records');
+  
+  cleanDOM(mainContainer);
+  const ownerRecord = document.createElement('span');
+  ownerRecord.innerHTML = `ownerAddress: ${ownerAddress}`;
+  const resolverRecord = document.createElement('span');
+  resolverRecord.innerHTML = `resolverAddress: ${resolverAddress}`;
+
+  mainContainer.appendChild(ownerRecord);
+  mainContainer.appendChild(resolverRecord);
+
+  Object.entries(records).map(([key, value]) => {
+    const recordSpan = document.createElement('span');
+    recordSpan.innerHTML = `${key} : ${value}`;
+    mainContainer.appendChild(recordSpan);
+  });
+
+  if (!records['crypto.BTC.address']) {
+    displayError('crypto.BTC.address: Record is not found', false);
+  }
+}
+
+async function resolve() {
+  const userInput = (<HTMLInputElement>document.getElementById("input")).value;
+  if (!userInput.endsWith(".zil")) {
+    displayError('domain is not supported');
+    return ;
+  }
+
+  const hash = namehash(userInput);  
+  const contractAddresses = await fetchZilliqa([UD_REGISTRY_CONTRACT, "records", [hash]]);
+  
+  if (contractAddresses.result == null) {
+    displayError('domain is not registered', true);
+    return ;
+  }
+  const [ownerAddress, resolverAddress] = await contractAddresses.result.records[hash].arguments;
+
+  if (resolverAddress === "0x0000000000000000000000000000000000000000") {
+    displayError('domain is not configured', true);
+    return ;
+  }
+
+  const records = await fetchZilliqa([
+    resolverAddress.replace("0x", ""),
+    "records",
+    []
+  ]).then(data => (data.result.records));
+  
+  displayResolution({resolverAddress, ownerAddress, records});
+}
+```
+
+{% hint style="info" %}
+If domain doesn't have a resolver address it will be set as **"0x0000000000000000000000000000000000000000"**
+
+_You  can check it with the domain: **paulalcock.zil**_
+{% endhint %}
+
+By this moment you can successfully resolve any .zil domain and show an appropriate error message for your users.
```

Comment:
What to do after the code is written? How to build the project? How file tree will look after the build?